src/DGtal/geometry/curves/AlphaThickSegmentComputer.h

```diff
@@ -0,0 +1,711 @@
+/**
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ **/
+
+#pragma once
+
+/**
+ * @file AlphaThickSegmentComputer.h
+ * @author Bertrand Kerautret (\c kerautre@loria.fr )
+ * LORIA (CNRS, UMR 7503), University of Nancy, France
+ * @author Alexandre Faure
+ * @author Fabien Feschet 
+ * @author Mohammad Said
+ * @author Jacques-Olivier Lachaud
+ *
+ * @date 2015/01/05
+ *
+ * Header file for module AlphaThickSegmentComputer.cpp
+ *
+ * This file is part of the DGtal library.
+ */
+
+#if defined(AlphaThickSegmentComputer_RECURSES)
+#error Recursive header files inclusion detected in AlphaThickSegmentComputer.h
+#else // defined(AlphaThickSegmentComputer_RECURSES)
+/** Prevents recursive inclusion of headers. */
+#define AlphaThickSegmentComputer_RECURSES
+
+#if !defined AlphaThickSegmentComputer_h
+/** Prevents repeated inclusion of headers. */
+#define AlphaThickSegmentComputer_h
+
+//////////////////////////////////////////////////////////////////////////////
+// Inclusions
+#include <iostream>
+#include "DGtal/base/Common.h"
+#include "DGtal/base/ReverseIterator.h"
+#include "DGtal/kernel/SpaceND.h"
+#include "DGtal/geometry/surfaces/ParallelStrip.h"
+//////////////////////////////////////////////////////////////////////////////
+
+namespace DGtal
+{
+
+/////////////////////////////////////////////////////////////////////////////
+// class AlphaThickSegmentComputer
+/**
+ * Description of class 'AlphaThickSegmentComputer' <p>
+ *
+ * \brief Aim: This class is devoted to the recognition of the alpha
+ * thick segment as described in \cite FaureTangential2008 . From a
+ * maximal diagonal alphaMax thickness, it allows to apply the recognition
+ * of a thick segment with the ability to take into accounts some
+ * noise. Moreover the segment can be detected from points not
```

Comment:
Moreover points of the segment may not be (digitally) connected and may have floating point coordinates. Connection is only given by the order of the points.